numpy/core/src/multiarray/einsum_sumprod.c.src

```diff
@@ -76,8 +73,15 @@
  *            0*5,
  *            0,0,1,0,
  *            0*3#
+ * #NPYV_CHK = 0*5,
+ *             0*5,
+ *             0, NPY_SIMD, NPY_SIMD_F64, 0,
+ *             0*3#
+ * #unroll_by = 0*5,
+ *              0*5,
+ *              0,2, 4, 0,
```

Comment:
Why 2 and 4? My naive model would be that smaller floats can be unrolled _more_, because they occupy less memory - but what you have here unrolls them less.

Reply:
Let's assume that our task is handle eight elements at a time in the array with the SIMD width of 128bits, The SIMD can deal with four floats in one loop, so we just need to unroll twice, but we can only deal with two doubles, which needs four unrolls.